hnn_core/viz.py

```diff
@@ -255,45 +255,55 @@ def plot_dipole(dpl, tmin=None, tmax=None, ax=None, layer='agg', decim=None,
         dpl = dpl + [average_dipoles(dpl)]
 
     scale_applied = dpl[0].scale_applied
-    for idx, dpl_trial in enumerate(dpl):
-        if dpl_trial.scale_applied != scale_applied:
-            raise RuntimeError('All dipoles must be scaled equally!')
-
-        if layer in dpl_trial.data.keys():
-
-            # extract scaled data and times
-            data, times = _get_plot_data_trange(dpl_trial.times,
-                                                dpl_trial.data[layer], tmin,
-                                                tmax)
-            if decim is not None:
-                data, times = _decimate_plot_data(decim, data, times)
-
-            if idx == len(dpl) - 1 and average:
-                # the average dpl
-                ax.plot(times, data, color='g', label="average", lw=1.5)
-            else:
-                alpha = 0.5 if average else 1.
-                ax.plot(times, data, color=color, alpha=alpha, lw=1.)
-
-    if average:
-        ax.legend()
-
-    ax.ticklabel_format(axis='both', scilimits=(-2, 3))
-    ax.set_xlabel('Time (ms)')
-    if scale_applied == 1:
-        ylabel = 'Dipole moment (nAm)'
-    else:
-        ylabel = 'Dipole moment\n(nAm ' +\
-            r'$\times$ {:.0f})'.format(scale_applied)
-    ax.set_ylabel(ylabel, multialignment='center')
-    if layer == 'agg':
-        title_str = 'Aggregate (L2 + L5)'
+    if isinstance(layer, str):
+        layer = [layer]
+        ax = [ax]
```

Comment:
separate
```suggestion
if isinstance(ax, plt.Axes)
        ax = [ax]
```

I think you might have to do something like this. Otherwise, providing `ax` will not work when layer is not str.